### PR TITLE
kpatch-build: respect CPPFLAGS

### PR DESCRIPTION
**Pull request by rudis**
Helps applying compiler flags from distributions, e.g. Debian.

### CODE REVIEW

**Comment by jpoimboe** [comment]
👍

**Comment by rudis** [comment]
Good idea, I've updated the patch to use make's implicit rules and also remove the compile rule which is also handled by make.

File: kpatch-build/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 include ../Makefile.inc
 
-CFLAGS  += -I../kmod/patch -Iinsn -Wall -g -Werror
-LDFLAGS += -lelf
+CFLAGS += -MMD -MP -I../kmod/patch -Iinsn -Wall -g
```
Comment (rudis) [inline kpatch-build/Makefile]: Sorry, no idea how that happened. Fixed version pushed.

**Comment by jpoimboe** [comment]
👍

**Comment by joe-lawrence** [comment]
Looks good, thanks @rudis

**Pull request merged**